tests/MemoryLeakWarningTest.cpp

```diff
@@ -54,6 +54,36 @@ static MemoryLeakWarningPlugin* memPlugin;
 static DummyReporter dummy;
 static TestMemoryAllocator* allocator;
 
+TEST_GROUP(MemoryLeakWarningLocalDetectorTest)
+{
+};
+
+TEST(MemoryLeakWarningLocalDetectorTest, localDetectorReturnsNewGlobalWhenNoneWasSet)
+{
+    MemoryLeakWarningPlugin memoryLeakWarningPlugin("TestMemoryLeakWarningPlugin", NULL);
+    CHECK(0 != memoryLeakWarningPlugin.getMemoryLeakDetector());
+}
+
+TEST(MemoryLeakWarningLocalDetectorTest, localDetectorIsTheOneSpecifiedInConstructor)
+{
+    MemoryLeakDetector localDetector(&dummy);
+    MemoryLeakDetector globalDetector(&dummy);
+    MemoryLeakWarningPlugin memoryLeakWarningPlugin("TestMemoryLeakWarningPlugin", &localDetector);
+    POINTERS_EQUAL(&localDetector, memoryLeakWarningPlugin.getMemoryLeakDetector());
+}
+
+IGNORE_TEST(MemoryLeakWarningLocalDetectorTest, localDetectorIsGlobalDetector)
```

Comment:
Help would be appreciated. This is a 'real' test case... Of course, I could cheat and not set the global detector at all and compare against getGlobalMemoryLeakDetector() ;-)